docs/scope-and-objectives.md

```diff
@@ -0,0 +1,139 @@
+Provider implementation: Existing Cluster API implementations consist of generic and infrastructure provider-specific logic. The infrastructure provider-specific logic is currently maintained in infrastructure provider repositories.
+- __Kubernetes-conformant__: A cluster that passes the Kubernetes conformance tests.
+- __Control plane__:
+   - __Self-provisioned__: A Kubernetes control plane consisting of pods or machines wholly managed by a single Cluster API deployment.
+   - __External__: A control plane offered and controlled by some system other than Cluster API (e.g., GKE, AKS, EKS, IKS).
+- __Core Add-ons__: Addons that are required to deploy a Kubernetes-conformant cluster: DNS, kube-proxy, CNI.
+- __Additional Add-ons__: Addons that are not required for a Kubernetes-conformant cluster (e.g. metrics/Heapster, Dashboard).
+- __Management cluster__: The cluster where one or more Cluster API providers run, and where resources (e.g. Machines) are stored.
+- __Workload cluster__: A cluster whose lifecycle is managed by the Management cluster.
+- __Operating system__: A generically understood combination of a kernel and system-level userspace interface, such as Linux or Windows, as opposed to a particular distribution.
+- __Manage a cluster__: create, scale, upgrade, destroy.
+
+## Motivation
+
+Kubernetes has a common set of APIs (see the [Kubernetes API Conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md)) to orchestrate containers regardless of deployment mechanism or cloud provider. Kubernetes also has APIs for handling some infrastructure, like load-balancers, ingress rules, or persistent volumes, but not for creating new machines.
+As a result, the deployment mechanisms that manage Kubernetes clusters each have unique APIs and implementations for how to handle lifecycle events like cluster creation or deletion, control plane upgrades, and node upgrades.
+
+### Goals
+
+- To manage the lifecycle (create, scale, upgrade, destroy) of Kubernetes-conformant clusters using a declarative API.
+- To work in different environments, both on-premises and in the cloud.
+- To define common operations, provide a default implementation, and provide the ability to swap out implementations for custom ones.
+- To reuse and integrate existing ecosystem components rather than duplicating their functionality (e.g. node-problem-detector, cluster autoscaler, SIG-Multi-cluster).
+- To provide a transition path for Kubernetes lifecycle products to adopt Cluster API incrementally. The result is a fragmentation of tooling, and we want to create a unified mechanism that gives users consistency around cluster management.
```

Comment:
It's not clear what "The result" refers to here.  Is it the transition path?  or the fact that these other tools exist, and the transition path is to help incrementally create the consistency?

Reply:
I believe the second sentence here can and should be deleted.

If we do want to add some additional context for this goal, we could always say something along the lines of:
"We want to provide a way for existing cluster lifecycle management tooling to be able to adopt Cluster API in a staged manner, allowing them to make the transition over the course of multiple releases, or even allowing them to only adopt a subset of Cluster API"

Reply:
OK, that makes sense, +1 to your proposed text

Reply:
Reworded a little in the last update, let me know if that works!